internal/http/services/owncloud/ocdav/move.go

```diff
@@ -26,203 +26,167 @@ import (
 
 	rpc "github.com/cs3org/go-cs3apis/cs3/rpc/v1beta1"
 	provider "github.com/cs3org/go-cs3apis/cs3/storage/provider/v1beta1"
-	"github.com/cs3org/reva/v2/internal/http/services/owncloud/ocdav/errors"
 	"github.com/cs3org/reva/v2/internal/http/services/owncloud/ocdav/net"
 	"github.com/cs3org/reva/v2/internal/http/services/owncloud/ocdav/spacelookup"
-	"github.com/cs3org/reva/v2/pkg/appctx"
 	"github.com/cs3org/reva/v2/pkg/errtypes"
 	rstatus "github.com/cs3org/reva/v2/pkg/rgrpc/status"
 	"github.com/cs3org/reva/v2/pkg/rhttp/router"
 	"github.com/cs3org/reva/v2/pkg/storagespace"
 	"github.com/cs3org/reva/v2/pkg/utils"
-	"github.com/rs/zerolog"
 )
 
-func (s *svc) handlePathMove(w http.ResponseWriter, r *http.Request, ns string) {
+func (s *svc) handlePathMove(w http.ResponseWriter, r *http.Request, ns string) (status int, err error) {
 	ctx, span := s.tracerProvider.Tracer(tracerName).Start(r.Context(), "move")
 	defer span.End()
 
 	if r.Body != http.NoBody {
-		w.WriteHeader(http.StatusUnsupportedMediaType)
-		return
+		return http.StatusUnsupportedMediaType, nil
 	}
 
 	srcPath := path.Join(ns, r.URL.Path)
 	dh := r.Header.Get(net.HeaderDestination)
 	baseURI := r.Context().Value(net.CtxKeyBaseURI).(string)
 	dstPath, err := net.ParseDestination(baseURI, dh)
 	if err != nil {
-		w.WriteHeader(http.StatusBadRequest)
-		return
+		return http.StatusBadRequest, err
 	}
 
 	for _, r := range nameRules {
 		if !r.Test(dstPath) {
-			w.WriteHeader(http.StatusBadRequest)
-			return
+			return http.StatusBadRequest, fmt.Errorf("path violates naming rules")
```

Comment:
can we have a reference in the error message where the rules are listed?

Reply:
Hm, no: if a user reports this error he has a broken client. It is still possible to search for this error string and then follow `nameRules` to its definition. Pointing to source code will only get outdated.

We could turn `nameRules` into a map with rules and error messages that tell the client exactly what he did wrong, but that is outside the scope of this PR